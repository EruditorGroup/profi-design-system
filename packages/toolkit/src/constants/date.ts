type ItemInfo<IDT = string> = {
  id: IDT;
  shortName: string;
  fullName: string;
};

export type Month = ItemInfo<
  | 'jan'
  | 'feb'
  | 'mar'
  | 'apr'
  | 'may'
  | 'jun'
  | 'jul'
  | 'aug'
  | 'sep'
  | 'oct'
  | 'nov'
  | 'dec'
>;

export const MONTHS: Month[] = [
  {id: 'jan', shortName: 'Янв', fullName: 'Январь'},
  {id: 'feb', shortName: 'Фев', fullName: 'Февраль'},
  {id: 'mar', shortName: 'Мар', fullName: 'Март'},
  {id: 'apr', shortName: 'Апр', fullName: 'Апрель'},
  {id: 'may', shortName: 'Май', fullName: 'Май'},
  {id: 'jun', shortName: 'Июн', fullName: 'Июнь'},
  {id: 'jul', shortName: 'Июл', fullName: 'Июль'},
  {id: 'aug', shortName: 'Авг', fullName: 'Август'},
  {id: 'sep', shortName: 'Сен', fullName: 'Сентябрь'},
  {id: 'oct', shortName: 'Окт', fullName: 'Октябрь'},
  {id: 'nov', shortName: 'Ноя', fullName: 'Ноябрь'},
  {id: 'dec', shortName: 'Дек', fullName: 'Декабрь'},
];
export const MONTH_NAMES = MONTHS.map((m) => m.fullName);

export type Weekday = ItemInfo<'mo' | 'tu' | 'we' | 'th' | 'fr' | 'sa' | 'su'>;

export const WEEKDAYS: Weekday[] = [
  {id: 'mo', shortName: 'Пн', fullName: 'Понедельник'},
  {id: 'tu', shortName: 'Вт', fullName: 'Вторник'},
  {id: 'we', shortName: 'Ср', fullName: 'Среда'},
  {id: 'th', shortName: 'Чт', fullName: 'Четверг'},
  {id: 'fr', shortName: 'Пт', fullName: 'Пятница'},
  {id: 'sa', shortName: 'Сб', fullName: 'Суббота'},
  {id: 'su', shortName: 'Вс', fullName: 'Воскресенье'},
];
export const DAYS_IN_WEEK = WEEKDAYS.length;
