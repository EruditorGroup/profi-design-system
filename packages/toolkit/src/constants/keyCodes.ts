export const BACKSPACE = 8;
export const TAB = 9;
export const ENTER = 13;
export const ESC = 27;
export const SPACE = 32;
export const LEFT_ARROW = 37;
export const UP_ARROW = 38;
export const RIGHT_ARROW = 39;
export const DOWN_ARROW = 40;
export const DELETE = 46;
export const NUMBER_ZERO = 48;
export const NUMBER_NINE = 57;
