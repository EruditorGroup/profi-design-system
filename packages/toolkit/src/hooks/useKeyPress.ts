import {useEffect} from 'react';
import usePersistCallback from './usePersistCallback';

export default function useKeyPress(
  targetKey: number,
  callback: () => void,
): void {
  const downHandler = usePersistCallback(({key, keyCode}: KeyboardEvent) => {
    const rightKey = isNaN(+key) ? keyCode : key; // key вместо кода может использовать название, в таком случае используем keycode
    if (String(rightKey) === String(targetKey)) {
      callback();
    }
  });

  useEffect(() => {
    window.addEventListener('keydown', downHandler);
    return () => {
      window.removeEventListener('keydown', downHandler);
    };
  }, [downHandler]);
}
