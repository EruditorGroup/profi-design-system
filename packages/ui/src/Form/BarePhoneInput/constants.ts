export type ICountryData = {
  placeholder: string;
  countryCode: ICountryCode;
  phoneCode: string;
  mask: string;
  subCodes: string[];
};
export type ICountryCode = string;

// страны присутствия
export const BASE_COUNTRIES: ICountryData[] = [
  {
    countryCode: 'ru',
    placeholder: '7 123 456 78-90',
    phoneCode: '7',
    mask: '# ###-###-##-##',
    subCodes: ['9'],
  },
  {
    countryCode: 'kz',
    placeholder: '7 123-456-78-90',
    phoneCode: '7',
    mask: '# ###-###-##-##',
    subCodes: [
      '7',
      '310',
      '311',
      '312',
      '313',
      '315',
      '318',
      '321',
      '324',
      '325',
      '326',
      '327',
      '336',
    ],
  },
  {
    countryCode: 'by',
    placeholder: '375 12 345-67-89',
    phoneCode: '375',
    mask: '### ## ###-##-##',
    subCodes: [],
  },
  {
    countryCode: 'ua',
    placeholder: '380 12 345-67-89',
    phoneCode: '380',
    mask: '### ## ###-##-##',
    subCodes: [],
  },
  {
    countryCode: 'af',
    placeholder: '93 99 99 999-9999',
    phoneCode: '93',
    mask: '## ## ###-####',
    subCodes: [],
  },
  {
    countryCode: 'al',
    placeholder: '355 999 9 999-9999',
    phoneCode: '355',
    mask: '### # ###-####',
    subCodes: [],
  },
  {
    countryCode: 'ad',
    placeholder: '376 999 999-999',
    phoneCode: '376',
    mask: '### ###-###',
    subCodes: [],
  },
  {
    countryCode: 'ar',
    placeholder: '54 99 99 9999-9999',
    phoneCode: '54',
    mask: '## ## ####-####',
    subCodes: [],
  },
  {
    countryCode: 'am',
    placeholder: '374 999 99 999999',
    phoneCode: '374',
    mask: '### ## ######',
    subCodes: [],
  },
  {
    countryCode: 'aw',
    placeholder: '297 999 999 9999',
    phoneCode: '297',
    mask: '### ### ####',
    subCodes: [],
  },
  {
    countryCode: 'au',
    placeholder: '61 99 9 9999 9999',
    phoneCode: '61',
    mask: '## # #### ####',
    subCodes: ['2', '3', '4', '7', '8', '02', '03', '04', '07', '08'],
  },
  {
    countryCode: 'at',
    placeholder: '43 99 9 9999999999',
    phoneCode: '43',
    mask: '## # ##########',
    subCodes: [],
  },
  {
    countryCode: 'az',
    placeholder: '994 999 99 999 99-99',
    phoneCode: '994',
    mask: '### ## ### ##-##',
    subCodes: [],
  },
  {
    countryCode: 'bs',
    placeholder: '1242 9 999-999-9999',
    phoneCode: '1242',
    mask: '# ###-###-####',
    subCodes: [],
  },
  {
    countryCode: 'bh',
    placeholder: '973 999 9999 9999',
    phoneCode: '973',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'bd',
    placeholder: '880 999 9-9999999',
    phoneCode: '880',
    mask: '### #-#######',
    subCodes: [],
  },
  {
    countryCode: 'bb',
    placeholder: '1246 9 999-999-9999',
    phoneCode: '1246',
    mask: '# ###-###-####',
    subCodes: [],
  },

  {
    countryCode: 'be',
    placeholder: '32 99 9 999 99 99',
    phoneCode: '32',
    mask: '## # ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'bz',
    placeholder: '501 999 999-9999',
    phoneCode: '501',
    mask: '### ###-####',
    subCodes: [],
  },
  {
    countryCode: 'bo',
    placeholder: '591 999 9999999',
    phoneCode: '591',
    mask: '### #######',
    subCodes: [],
  },
  {
    countryCode: 'ba',
    placeholder: '387 999 99 999-999',
    phoneCode: '387',
    mask: '### ## ###-###',
    subCodes: [],
  },
  {
    countryCode: 'br',
    placeholder: '55 99 99 9999-9999',
    phoneCode: '55',
    mask: '## ## ####-####',
    subCodes: [],
  },
  {
    countryCode: 'bn',
    placeholder: '673 999 999 9999',
    phoneCode: '673',
    mask: '### ### ####',
    subCodes: [],
  },
  {
    countryCode: 'bg',
    placeholder: '359 999 9 999 9999',
    phoneCode: '359',
    mask: '### # ### ####',
    subCodes: [],
  },
  {
    countryCode: 'kh',
    placeholder: '855 999 99 999 999',
    phoneCode: '855',
    mask: '### ## ### ###',
    subCodes: [],
  },
  {
    countryCode: 'ca',
    placeholder: '1 9 999-999-9999',
    phoneCode: '1',
    mask: '# ###-###-####',
    subCodes: [
      '204',
      '226',
      '236',
      '249',
      '250',
      '289',
      '306',
      '343',
      '365',
      '367',
      '387',
      '403',
      '416',
      '418',
      '431',
      '437',
      '438',
      '450',
      '506',
      '514',
      '519',
      '548',
      '579',
      '581',
      '587',
      '604',
      '613',
      '639',
      '647',
      '672',
      '705',
      '709',
      '742',
      '778',
      '780',
      '782',
      '807',
      '819',
      '825',
      '867',
      '873',
      '902',
      '905',
    ],
  },
  {
    countryCode: 'cl',
    placeholder: '56 99 9 9999 9999',
    phoneCode: '56',
    mask: '## # #### ####',
    subCodes: [],
  },
  {
    countryCode: 'cn',
    placeholder: '86 99 99 99999',
    phoneCode: '86',
    mask: '## ## #####',
    subCodes: [],
  },
  {
    countryCode: 'co',
    placeholder: '57 99 9 9999999',
    phoneCode: '57',
    mask: '## # #######',
    subCodes: [],
  },
  {
    countryCode: 'hr',
    placeholder: '385 999 9 9999 999',
    phoneCode: '385',
    mask: '### # #### ###',
    subCodes: [],
  },
  {
    countryCode: 'cu',
    placeholder: '53 99 9 9999999',
    phoneCode: '53',
    mask: '## # #######',
    subCodes: [],
  },
  {
    countryCode: 'cy',
    placeholder: '357 999 99 999999',
    phoneCode: '357',
    mask: '### ## ######',
    subCodes: [],
  },
  {
    countryCode: 'cz',
    placeholder: '420 999 999 999 999',
    phoneCode: '420',
    mask: '### ### ### ###',
    subCodes: [],
  },
  {
    countryCode: 'dk',
    placeholder: '45 99 99 99 99 99',
    phoneCode: '45',
    mask: '## ## ## ## ##',
    subCodes: [],
  },
  {
    countryCode: 'dm',
    placeholder: '1767 9 999-999-9999',
    phoneCode: '1767',
    mask: '# ###-###-####',
    subCodes: [],
  },
  {
    countryCode: 'do',
    placeholder: '1 9 999-999-9999',
    phoneCode: '1',
    mask: '# ###-###-####',
    subCodes: ['809', '829', '849'],
  },
  {
    countryCode: 'ec',
    placeholder: '593 999 99 999 9999',
    phoneCode: '593',
    mask: '### ## ### ####',
    subCodes: [],
  },
  {
    countryCode: 'eg',
    placeholder: '20 99 9 99999999',
    phoneCode: '20',
    mask: '## # ########',
    subCodes: [],
  },
  {
    countryCode: 'sv',
    placeholder: '503 999 9999 9999',
    phoneCode: '503',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'ee',
    placeholder: '372 999 99 999999',
    phoneCode: '372',
    mask: '### ## ######',
    subCodes: [],
  },
  {
    countryCode: 'fj',
    placeholder: '679 999 999 9999',
    phoneCode: '679',
    mask: '### ### ####',
    subCodes: [],
  },
  {
    countryCode: 'fi',
    placeholder: '358 999 999 99999',
    phoneCode: '358',
    mask: '### ### #####',
    subCodes: [],
  },
  {
    countryCode: 'fr',
    placeholder: '33 99 9 99 99 99 99',
    phoneCode: '33',
    mask: '## # ## ## ## ##',
    subCodes: [],
  },
  {
    countryCode: 'gf',
    placeholder: '594 999 999 99 99 99',
    phoneCode: '594',
    mask: '### ### ## ## ##',
    subCodes: [],
  },
  {
    countryCode: 'pf',
    placeholder: '689 999 99 99 99 99',
    phoneCode: '689',
    mask: '### ## ## ## ##',
    subCodes: [],
  },
  {
    countryCode: 'ge',
    placeholder: '995 999 99 999 99 99',
    phoneCode: '995',
    mask: '### ## ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'de',
    placeholder: '49 99 999 9 9999999',
    phoneCode: '49',
    mask: '## ### # #######',
    subCodes: [],
  },
  {
    countryCode: 'gr',
    placeholder: '30 99 99 9999 9999',
    phoneCode: '30',
    mask: '## ## #### ####',
    subCodes: [],
  },
  {
    countryCode: 'gd',
    placeholder: '1473 9 999-999-9999',
    phoneCode: '1473',
    mask: '# ###-###-####',
    subCodes: [],
  },

  {
    countryCode: 'gp',
    placeholder: '590 999 999 99 99 99',
    phoneCode: '590',
    mask: '### ### ## ## ##',
    subCodes: [],
  },
  {
    countryCode: 'gu',
    placeholder: '1671 9 999-999-9999',
    phoneCode: '1671',
    mask: '# ###-###-####',
    subCodes: [],
  },
  {
    countryCode: 'gt',
    placeholder: '502 999 9999 9999',
    phoneCode: '502',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'ht',
    placeholder: '509 999 9999 9999',
    phoneCode: '509',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'hn',
    placeholder: '504 999 9999-9999',
    phoneCode: '504',
    mask: '### ####-####',
    subCodes: [],
  },
  {
    countryCode: 'hk',
    placeholder: '852 999 9999 9999',
    phoneCode: '852',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'hu',
    placeholder: '36 99 99 999 9999',
    phoneCode: '36',
    mask: '## ## ### ####',
    subCodes: [],
  },
  {
    countryCode: 'is',
    placeholder: '354 999 999 9999',
    phoneCode: '354',
    mask: '### ### ####',
    subCodes: [],
  },
  {
    countryCode: 'in',
    placeholder: '91 99 999 999 9999',
    phoneCode: '91',
    mask: '## ### ### ####',
    subCodes: [],
  },
  {
    countryCode: 'id',
    placeholder: '62 99 999 9999999',
    phoneCode: '62',
    mask: '## ### #######',
    subCodes: [],
  },
  {
    countryCode: 'ir',
    placeholder: '98 99 9999999999',
    phoneCode: '98',
    mask: '## ##########',
    subCodes: [],
  },
  {
    countryCode: 'iq',
    placeholder: '964 999 999 999 9999',
    phoneCode: '964',
    mask: '### ### ### ####',
    subCodes: [],
  },
  {
    countryCode: 'ie',
    placeholder: '353 999 99 999 999',
    phoneCode: '353',
    mask: '### ## ### ###',
    subCodes: [],
  },
  {
    countryCode: 'il',
    placeholder: '972 999 9-999-99999',
    phoneCode: '972',
    mask: '### #-###-#####',
    subCodes: [],
  },
  {
    countryCode: 'it',
    placeholder: '39 99 99 99999999',
    phoneCode: '39',
    mask: '## ## ########',
    subCodes: [],
  },
  {
    countryCode: 'jm',
    placeholder: '1876 9 999-999-9999',
    phoneCode: '1876',
    mask: '# ###-###-####',
    subCodes: [],
  },
  {
    countryCode: 'jp',
    placeholder: '81 99 999-99-9999',
    phoneCode: '81',
    mask: '## ###-##-####',
    subCodes: [],
  },
  {
    countryCode: 'jo',
    placeholder: '962 999 9 999 9999',
    phoneCode: '962',
    mask: '### # ### ####',
    subCodes: [],
  },
  {
    countryCode: 'xk',
    placeholder: '383 999 99 9999999999',
    phoneCode: '383',
    mask: '### ## ##########',
    subCodes: [],
  },
  {
    countryCode: 'kw',
    placeholder: '965 999 9999 9999',
    phoneCode: '965',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'kg',
    placeholder: '996 999 999 999 999',
    phoneCode: '996',
    mask: '### ### ### ###',
    subCodes: [],
  },
  {
    countryCode: 'la',
    placeholder: '856 999 99 99 999 999',
    phoneCode: '856',
    mask: '### ## ## ### ###',
    subCodes: [],
  },
  {
    countryCode: 'lv',
    placeholder: '371 999 99 999 999',
    phoneCode: '371',
    mask: '### ## ### ###',
    subCodes: [],
  },
  {
    countryCode: 'lb',
    placeholder: '961 999 9 999 999',
    phoneCode: '961',
    mask: '### # ### ###',
    subCodes: [],
  },
  {
    countryCode: 'li',
    placeholder: '423 999 99 999 99 99',
    phoneCode: '423',
    mask: '### ## ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'lt',
    placeholder: '370 999 999 99999',
    phoneCode: '370',
    mask: '### ### #####',
    subCodes: [],
  },
  {
    countryCode: 'lu',
    placeholder: '352 999 99 999',
    phoneCode: '352',
    mask: '### ## ###',
    subCodes: [],
  },
  {
    countryCode: 'mo',
    placeholder: '853 999 9999 9999',
    phoneCode: '853',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'mk',
    placeholder: '389 999 9 999 9999',
    phoneCode: '389',
    mask: '### # ### ####',
    subCodes: [],
  },
  {
    countryCode: 'my',
    placeholder: '60 99 9-9999 9999',
    phoneCode: '60',
    mask: '## #-#### ####',
    subCodes: [],
  },

  {
    countryCode: 'mv',
    placeholder: '960 999 999-9999',
    phoneCode: '960',
    mask: '### ###-####',
    subCodes: [],
  },
  {
    countryCode: 'mt',
    placeholder: '356 999 9999 9999',
    phoneCode: '356',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'mx',
    placeholder: '52 99 99 9999 9999',
    phoneCode: '52',
    mask: '## ## #### ####',
    subCodes: ['55', '81', '33', '656', '664', '998', '774', '229'],
  },
  {
    countryCode: 'md',
    placeholder: '373 999 99 999 999',
    phoneCode: '373',
    mask: '### ## ### ###',
    subCodes: [],
  },
  {
    countryCode: 'mn',
    placeholder: '976 999 99 99 9999',
    phoneCode: '976',
    mask: '### ## ## ####',
    subCodes: [],
  },
  {
    countryCode: 'me',
    placeholder: '382 999 99 999 999',
    phoneCode: '382',
    mask: '### ## ### ###',
    subCodes: [],
  },
  {
    countryCode: 'mm',
    placeholder: '95 99 9 999 999',
    phoneCode: '95',
    mask: '## # ### ###',
    subCodes: [],
  },
  {
    countryCode: 'np',
    placeholder: '977 999 9-9999999',
    phoneCode: '977',
    mask: '### #-#######',
    subCodes: [],
  },
  {
    countryCode: 'nl',
    placeholder: '31 99 99 999 9999',
    phoneCode: '31',
    mask: '## ## ### ####',
    subCodes: [],
  },
  {
    countryCode: 'nz',
    placeholder: '64 99 9-999 9999',
    phoneCode: '64',
    mask: '## #-### ####',
    subCodes: [],
  },
  {
    countryCode: 'ni',
    placeholder: '505 999 9999 9999',
    phoneCode: '505',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'no',
    placeholder: '47 99 99 99 99 99',
    phoneCode: '47',
    mask: '## ## ## ## ##',
    subCodes: [],
  },
  {
    countryCode: 'om',
    placeholder: '968 999 999 99999',
    phoneCode: '968',
    mask: '### ### #####',
    subCodes: [],
  },
  {
    countryCode: 'pk',
    placeholder: '92 99 99 99999999',
    phoneCode: '92',
    mask: '## ## ########',
    subCodes: [],
  },
  {
    countryCode: 'ps',
    placeholder: '970 999 9-999-9999',
    phoneCode: '970',
    mask: '### #-###-####',
    subCodes: [],
  },
  {
    countryCode: 'py',
    placeholder: '595 999 999 99999',
    phoneCode: '595',
    mask: '### ### #####',
    subCodes: [],
  },
  {
    countryCode: 'pe',
    placeholder: '51 99 99 999999',
    phoneCode: '51',
    mask: '## ## ######',
    subCodes: [],
  },
  {
    countryCode: 'ph',
    placeholder: '63 99 9 999 9999',
    phoneCode: '63',
    mask: '## # ### ####',
    subCodes: [],
  },
  {
    countryCode: 'pl',
    placeholder: '48 99 99 999 99 99',
    phoneCode: '48',
    mask: '## ## ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'pt',
    placeholder: '351 999 9 9999 9999',
    phoneCode: '351',
    mask: '### # #### ####',
    subCodes: [],
  },
  {
    countryCode: 'pr',
    placeholder: '1 9 999 999 9999',
    phoneCode: '1',
    mask: '# ### ### ####',
    subCodes: ['787', '939'],
  },
  {
    countryCode: 'qa',
    placeholder: '974 999 9999 9999',
    phoneCode: '974',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'ro',
    placeholder: '40 99 99 999 9999',
    phoneCode: '40',
    mask: '## ## ### ####',
    subCodes: [],
  },
  {
    countryCode: 'sm',
    placeholder: '378 999 9999 999999',
    phoneCode: '378',
    mask: '### #### ######',
    subCodes: [],
  },
  {
    countryCode: 'sa',
    placeholder: '966 999 9999 99999',
    phoneCode: '966',
    mask: '### #### #####',
    subCodes: [],
  },
  {
    countryCode: 'rs',
    placeholder: '381 999 99 9999999',
    phoneCode: '381',
    mask: '### ## #######',
    subCodes: [],
  },
  {
    countryCode: 'sg',
    placeholder: '65 99 9999 9999',
    phoneCode: '65',
    mask: '## #### ####',
    subCodes: [],
  },
  {
    countryCode: 'sk',
    placeholder: '421 999 99/999 99 99',
    phoneCode: '421',
    mask: '### ##/### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'si',
    placeholder: '386 999 9 999 99 99',
    phoneCode: '386',
    mask: '### # ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'sb',
    placeholder: '677 999 99999',
    phoneCode: '677',
    mask: '### #####',
    subCodes: [],
  },
  {
    countryCode: 'kr',
    placeholder: '82 99 9999-9999',
    phoneCode: '82',
    mask: '## ####-####',
    subCodes: [],
  },
  {
    countryCode: 'es',
    placeholder: '34 99 999 99 99 99',
    phoneCode: '34',
    mask: '## ### ## ## ##',
    subCodes: [],
  },
  {
    countryCode: 'lk',
    placeholder: '94 99 999 999 999',
    phoneCode: '94',
    mask: '## ### ### ###',
    subCodes: [],
  },
  {
    countryCode: 'se',
    placeholder: '46 99 99 999 99 99',
    phoneCode: '46',
    mask: '## ## ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'ch',
    placeholder: '41 99 99 999 99 99',
    phoneCode: '41',
    mask: '## ## ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'sy',
    placeholder: '963 999 99 999 9999',
    phoneCode: '963',
    mask: '### ## ### ####',
    subCodes: [],
  },
  {
    countryCode: 'tw',
    placeholder: '886 999 9 9999 9999',
    phoneCode: '886',
    mask: '### # #### ####',
    subCodes: [],
  },
  {
    countryCode: 'tj',
    placeholder: '992 999 999 99 9999',
    phoneCode: '992',
    mask: '### ### ## ####',
    subCodes: [],
  },
  {
    countryCode: 'th',
    placeholder: '66 99 9 999 9999',
    phoneCode: '66',
    mask: '## # ### ####',
    subCodes: [],
  },
  {
    countryCode: 'tr',
    placeholder: '90 99 999 999 99 99',
    phoneCode: '90',
    mask: '## ### ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 'tm',
    placeholder: '993 999 99 999999',
    phoneCode: '993',
    mask: '### ## ######',
    subCodes: [],
  },
  {
    countryCode: 'ae',
    placeholder: '971 999 9 999 9999',
    phoneCode: '971',
    mask: '### # ### ####',
    subCodes: [],
  },
  {
    countryCode: 'gb',
    placeholder: '44 99 999 999 9999',
    phoneCode: '44',
    mask: '## ### ### ####',
    subCodes: [],
  },
  {
    countryCode: 'us',
    placeholder: '1 9 999-999-9999',
    phoneCode: '1',
    mask: '# ###-###-####',
    subCodes: [
      '20',
      '210',
      '212',
      '213',
      '214',
      '215',
      '216',
      '217',
      '218',
      '219',
      '220',
      '223',
      '224',
      '225',
      '226',
      '228',
      '229',
      '231',
      '234',
      '236',
      '239',
      '240',
      '242',
      '246',
      '248',
      '250',
      '251',
      '252',
      '253',
      '254',
      '256',
      '260',
      '262',
      '264',
      '267',
      '268',
      '269',
      '270',
      '272',
      '276',
      '279',
      '281',
      '284',
      '289',
      '30',
      '310',
      '312',
      '31',
      '320',
      '321',
      '323',
      '325',
      '326',
      '330',
      '331',
      '332',
      '334',
      '336',
      '337',
      '339',
      '340',
      '341',
      '343',
      '345',
      '346',
      '347',
      '351',
      '352',
      '360',
      '361',
      '364',
      '365',
      '367',
      '380',
      '385',
      '386',
      '40',
      '410',
      '412',
      '413',
      '414',
      '415',
      '417',
      '418',
      '419',
      '423',
      '424',
      '425',
      '430',
      '431',
      '432',
      '434',
      '435',
      '437',
      '438',
      '440',
      '441',
      '442',
      '443',
      '445',
      '458',
      '463',
      '459',
      '470',
      '473',
      '475',
      '478',
      '479',
      '480',
      '50',
      '51',
      '520',
      '530',
      '531',
      '534',
      '539',
      '540',
      '541',
      '548',
      '551',
      '559',
      '561',
      '562',
      '563',
      '564',
      '567',
      '570',
      '573',
      '574',
      '575',
      '579',
      '580',
      '581',
      '585',
      '586',
      '587',
      '60',
      '610',
      '612',
      '614',
      '615',
      '616',
      '617',
      '618',
      '619',
      '620',
      '623',
      '626',
      '629',
      '630',
      '631',
      '636',
      '640',
      '641',
      '646',
      '650',
      '651',
      '657',
      '658',
      '659',
      '660',
      '661',
      '662',
      '664',
      '667',
      '670',
      '671',
      '673',
      '678',
      '680',
      '681',
      '682',
      '684',
      '689',
      '70',
      '712',
      '713',
      '714',
      '715',
      '716',
      '717',
      '718',
      '719',
      '720',
      '724',
      '725',
      '726',
      '727',
      '731',
      '732',
      '734',
      '737',
      '740',
      '743',
      '747',
      '754',
      '757',
      '758',
      '760',
      '762',
      '763',
      '765',
      '767',
      '769',
      '770',
      '772',
      '773',
      '774',
      '775',
      '778',
      '779',
      '781',
      '782',
      '784',
      '785',
      '786',
      '787',
      '80',
      '810',
      '812',
      '813',
      '814',
      '815',
      '816',
      '817',
      '818',
      '819',
      '820',
      '825',
      '828',
      '829',
      '830',
      '831',
      '832',
      '838',
      '839',
      '843',
      '845',
      '847',
      '848',
      '849',
      '850',
      '854',
      '856',
      '857',
      '858',
      '859',
      '860',
      '862',
      '863',
      '864',
      '865',
      '867',
      '868',
      '869',
      '870',
      '872',
      '873',
      '876',
      '878',
      '90',
      '910',
      '912',
      '913',
      '914',
      '915',
      '916',
      '918',
      '919',
      '920',
      '925',
      '928',
      '931',
      '934',
      '936',
      '937',
      '938',
      '940',
      '941',
      '947',
      '949',
      '951',
      '952',
      '954',
      '956',
      '959',
      '970',
      '971',
      '972',
      '973',
      '978',
      '979',
      '980',
      '985',
      '986',
      '989',
      '907',
    ],
  },
  {
    countryCode: 'uy',
    placeholder: '598 999 9999 9999',
    phoneCode: '598',
    mask: '### #### ####',
    subCodes: [],
  },
  {
    countryCode: 'uz',
    placeholder: '998 999 99 999 99 99',
    phoneCode: '998',
    mask: '### ## ### ## ##',
    subCodes: [],
  },
  {
    countryCode: 've',
    placeholder: '58 99 999-9999999',
    phoneCode: '58',
    mask: '## ###-#######',
    subCodes: [],
  },
  {
    countryCode: 'vn',
    placeholder: '84 99 99 9999 9999',
    phoneCode: '84',
    mask: '## ## #### ####',
    subCodes: [],
  },
];

export const ELITE_PATTERNS = ['812', '800'];
