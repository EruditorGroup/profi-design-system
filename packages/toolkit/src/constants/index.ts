import * as KEY_CODES from './keyCodes';
export * from './date';
export * from './time';
export {KEY_CODES};
