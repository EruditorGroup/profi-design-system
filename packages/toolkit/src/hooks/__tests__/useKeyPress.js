import useKeyPress from '../useKeyPress';
import {renderHook} from '@testing-library/react-hooks';
import {ESC} from '../../constants/keyCodes';

jest.mock('../usePersistCallback', () => (fn) => fn);

jest.unmock('../useKeyPress');
jest.unmock('../../constants/keyCodes');

describe('useKeyPress', () => {
  it('subscribe', () => {
    const callback = jest.fn();

    window.addEventListener = jest.fn();
    renderHook(() => useKeyPress(ESC, callback));
    expect(callback).not.toBeCalled();

    const lastEvents = window.addEventListener.mock.calls.filter(
      ([event]) => event === 'keydown',
    );

    expect(lastEvents).toHaveLength(1);
  });

  it('unsubscribe', () => {
    const callback = jest.fn();
    const hook = renderHook(() => useKeyPress(ESC, callback));
    window.removeEventListener = jest.fn();
    hook.unmount();

    const lastEvents = window.removeEventListener.mock.calls.filter(
      ([event]) => event === 'keydown',
    );

    expect(lastEvents).toHaveLength(1);
  });

  it('event fire', () => {
    const evCollection = [];
    window.addEventListener = jest.fn((key, fn) => {
      evCollection[key] = fn;
    });

    const escapeCallback = jest.fn();
    renderHook(() => useKeyPress(ESC, escapeCallback));
    expect(evCollection.keydown).toMatchSnapshot();

    evCollection.keydown(new KeyboardEvent('keydown', {key: ESC}));

    expect(escapeCallback).toBeCalled();
  });

  it('event fire when key = keyName', () => {
    const evCollection = [];
    window.addEventListener = jest.fn((key, fn) => {
      evCollection[key] = fn;
    });

    const escapeCallback = jest.fn();
    renderHook(() => useKeyPress(ESC, escapeCallback));
    expect(evCollection.keydown).toMatchSnapshot();

    evCollection.keydown(
      new KeyboardEvent('keydown', {key: 'Escape', keyCode: ESC}),
    );

    expect(escapeCallback).toBeCalled();
  });
});
