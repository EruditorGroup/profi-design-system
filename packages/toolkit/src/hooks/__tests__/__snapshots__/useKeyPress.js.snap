// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useKeyPress event fire 1`] = `[Function]`;

exports[`useKeyPress event fire when key = keyName 1`] = `[Function]`;
